src/Type/DynamicFunctionParameterOutTypeExtension.php

```diff
@@ -0,0 +1,17 @@
+<?php declare(strict_types = 1);
+
+namespace PHPStan\Type;
+
+use PhpParser\Node\Expr\FuncCall;
+use PHPStan\Analyser\Scope;
+use PHPStan\Reflection\FunctionReflection;
+use PHPStan\Reflection\ParameterReflection;
+
+interface DynamicFunctionParameterOutTypeExtension
```

Comment:
These interfaces need to be marked with `@api` and a description how to register them.